lib/navigation/preprocess_keyence.hpp

```diff
@@ -4,17 +4,22 @@
 #include <cstdint>
 
 #include "core/types.hpp"
+#include "core/logger.hpp"
 
 namespace hyped::navigation {
 
 class KeyencePreprocessor {
  public:
-  KeyencePreprocessor();
+ /**
+  * @brief Construct a new Keyence Preprocessor object
+  */
+  KeyencePreprocessor(hyped::core::ILogger &logger);
 
   SensorChecks checkKeyenceAgrees(const core::KeyenceData &keyence_data);
 
  private:
-  bool is_keyence_agreeing_;
+  hyped::core::ILogger &logger_;
```

Comment:
`&log_` seems to be the codebase default?

Reply:
logger_  was used in adc class in io, which is where I pulled this code from . Checked their other files and log_ is used, so I'll change this. @SnickeyX you'll want to know this.